tests/sos_tests.py

```diff
@@ -0,0 +1,464 @@
+
+    @property
+    def tmpdir(self):
+        if not self._tmpdir:
+            self._tmpdir = os.getenv('AVOCADO_TESTS_COMMON_TMPDIR') + self.klass_name
+        return self._tmpdir
+
+    def generate_sysinfo(self):
+        """Collects some basic information about the system for later reference
+        in individual tests
+        """
+        sysinfo = {}
+
+        # get kernel modules
+        mods = []
+        _out = process.run('lsmod').stdout.decode()
+        for line in _out.splitlines()[1:]:
+            mods.append(line.split()[0])
+        sysinfo['modules'] = mods
+
+        return sysinfo
+
+    def _write_file_to_tmpdir(self, fname, content):
+        """Write the given content to fname within the test's tmpdir
+        """
+        fname = os.path.join(self.tmpdir, fname)
+        if isinstance(content, bytes):
+            content = content.decode()
+        with open(fname, 'w') as wfile:
+            wfile.write(content)
+
+    def read_file_from_tmpdir(self, fname):
+        fname = os.path.join(self.tmpdir, fname)
+        with open(fname, 'r') as tfile:
+            return tfile.read()
+        return ''
+
+    def _write_sysinfo(self, fname):
+        """Get the current state of sysinfo and write it into our shared
+        tempdir so it can be loaded in setUp() later
+
+        :param fname:  The name of the file to be written in the tempdir
+        :type fname: ``str``
+        """
+        sysinfo = self.generate_sysinfo()
+        self._write_file_to_tmpdir(fname, json.dumps(sysinfo))
+
+    def _read_sysinfo(self, fname):
+        sysinfo = {}
+        content = self.read_file_from_tmpdir(fname)
+        if content:
+            sysinfo = json.loads(content)
+        return sysinfo
+
+    def set_pre_sysinfo(self):
+        self._write_sysinfo('pre_sysinfo')
+
+    def get_pre_sysinfo(self):
+        return self._read_sysinfo('pre_sysinfo')
+
+    def set_post_sysinfo(self):
+        self._write_sysinfo('post_sysinfo')
+
+    def get_post_sysinfo(self):
+        return self._read_sysinfo('post_sysinfo')
+
+    def get_sysinfo(self):
+        sinfo = {
+            'pre': self.get_pre_sysinfo(),
+            'post': self.get_post_sysinfo()
+        }
+        return sinfo
+
+    def assertFileExists(self, fname):
+        """Asserts that fname exists on the filesystem"""
+        assert os.path.exists(fname), "%s does not exist" % fname
+
+    def assertFileNotExists(self, fname):
+        """Asserts that fname does not exist on the filesystem"""
+        assert not os.path.exists(fname), "%s exists" % fname
+        
+
+class BaseSoSReportTest(BaseSoSTest):
+    """This is the class to use for building sos report tests with.
+
+    An instance of this test is expected to set at minimum a ``sos_cmd`` class
+    attribute that represets the options handed to a specific execution of an
+    sos command. This should be anything following ``sos report --batch``.
+
+    :avocado: disable
```

Comment:
I don't think this is needed, because there are no `test_` methods defined in this class.

Reply:
Ack.